src/twisted/application/twist/test/test_twist.py

```diff
@@ -191,3 +191,80 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    """
+    Tests to verify that the Twist script takes the expected actions related
+    to signals and the reactor.
+    """
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def fakeExitWithSignal(sig):
+            """
+            Fake to capture whether L{twisted.application._exitWithSignal
+            was called.
+
+            @param sig: Signal value
+            @type sig: C{int}
+            """
+            self.exitWithSignalCalled = True
+
+        self.patch(_twist, '_exitWithSignal', fakeExitWithSignal)
+
+        def startLogging(_):
+            """
+            Prevent Runner from adding new log observers or other
+            tests outside this module will fail.
+
+            @param _: Unused self param
+            """
+            pass
+
+        self.patch(Runner, 'startLogging', startLogging)
```

Comment:
`patch` on a global object is always a form of technical debt.

Is there any possible way you can inject a `Runner` instance into `Twist`?  This could be a follow-up if you don't want to set up this plumbing in this PR, I just don't want to lose track of it.